### PR TITLE
Provision Guac Collectsub and replace registry user/password variables with envars

### PR DESCRIPTION
**Pull request by gildub**
1. Provision Guac Collectsub pod from deployment manifest requiring a user provided secret manifest containing the TLS certificate.
2. Removes user/password variables from `play.yaml` file. Avoids any hard coded secret values in configuration files.
3. Allows auth access token for image registry - See https://access.redhat.com/articles/3560571 for more details.
3. Updates README.md accordingly

The following envars are now available for providing sensitive information :
* Subscription Registration user and password :
  - TPA_SINGLE_NODE_REGISTRATION_USERNAME 
  - TPA_SINGLE_NODE_REGISTRATION_PASSWORD
* Image Registry user and password :
  - TPA_SINGLE_NODE_REGISTRY_USERNAME
  - TPA_SINGLE_NODE_REGISTRY_PASSWORD
* Path for Guac Collectsub TLS secret manifest file: 
  - TPA_SINGLE_NODE_GUAC_CSUB_SECRET_FILE

### CODE REVIEW

**Review by desmax74 — APPROVED** [review]

File: vm-testing/rhel9-vm/Vagrantfile [hunk 1]
```diff
@@ -4,15 +4,13 @@
 Vagrant.configure("2") do |config|
     # https://github.com/projectatomic/adb-vagrant-registration
     config.registration.unregister_on_halt = false
-    #config.registration.username = '<your Red Hat username>'
-    #config.registration.password = '<your Red Hat password>'
-    config.registration.username = ENV['TPA_SINGLE_NODE_REGISTRY_USERNAME']
-    config.registration.password = ENV['TPA_SINGLE_NODE_REGISTRY_PASSWORD']
+    config.registration.username = ENV['TPA_SINGLE_NODE_REGISTRATION_USERNAME']
```
Comment (desmax74) [inline vm-testing/rhel9-vm/Vagrantfile:7]: Are the same credentials of the registry ? If yes why duplicate ?

File: README.md [hunk 2]
```diff
@@ -58,36 +63,51 @@ ansible_ssh_pass=vargrant
...
 ```
-3. Add your RH credentials, choose one:
- - as a env variables `SUBSCRIPTION_USERNAME` and `SUBSCRIPTION_PASSWORD` 
- - in the Vagrant file `config.registration.username` and `config.registration.password`
- - during the vagrant up in the registration prompt
 
+3. Add the subscription, registry and certificates information :
+
+- For Red Hat subscription define :
+  export TPA_SINGLE_NODE_REGISTRATION_USERNAME=<Your Red Hat subscription username>
```
Comment (desmax74) [inline README.md:77]: Are the same the credentials for subscription and tò access the registry ?

File: README.md [hunk 3]
```diff
@@ -58,36 +63,51 @@ ansible_ssh_pass=vargrant
...
+  export TPA_SINGLE_NODE_REGISTRATION_PASSWORD=<Your Red Hat subscription password>
+- For Red Hat image registry define :
+  export TPA_SINGLE_NODE_REGISTRY_USERNAME=<Your Red Hat image registry username>
+  export TPA_SINGLE_NODE_REGISTRY_PASSWORD=<Your Red Hat image registry password>
+
+# Todo - Remove ?
+
+Alternatively vagrant will prompt for the registration username and password.
+
+Provide TLS certificates :
```
Comment (desmax74) [inline README.md:87]: Add in the development section an example about how to create and where place or a file to be loaded automatically by vagrant

File: vm-testing/rhel9-vm/Vagrantfile (hunk 1 above)
Comment (gildub) [reply vm-testing/rhel9-vm/Vagrantfile:7]: Long story short it allows for auth access token (see article https://access.redhat.com/articles/3560571 which I've also added in PR's description).
When using token the Registry user creds are different than the Registration (RHN) user/password.

File: README.md (hunk 2 above)
Comment (gildub) [reply README.md:77]: As I explained above, they could be the same but when using auth access token they are different.

File: README.md [hunk 4]
```diff
@@ -42,12 +44,15 @@ ansible-galaxy collection install -r requirements.yml
...
 ## Provision
+
 #### https://developer.hashicorp.com/vagrant/docs/provisioning/ansible
+
 #### https://docs.ansible.com/ansible/2.9/scenario_guides/guide_vagrant.html
 
 In order to deploy Trustification on a RHEL 9.2+ VM:
 
-1. Create an `inventory.ini` file  in the project with a single VM in the `trustification` group:
+1. Create an `inventory.ini` file in the project with a single VM in the `trustification` group:
```
Comment (i386x) [inline README.md:54]: Just note: Ansible also supports inventories generated by executable, a.k.a dynamic inventories: https://docs.ansible.com/ansible/latest/inventory_guide/intro_dynamic_inventory.html#intro-dynamic-inventory

File: README.md [hunk 5]
```diff
@@ -58,36 +63,51 @@ ansible_ssh_pass=vargrant
...
 [defaults]
 inventory = ./inventory.ini
 host_key_checking = False
 ```
-3. Add your RH credentials, choose one:
- - as a env variables `SUBSCRIPTION_USERNAME` and `SUBSCRIPTION_PASSWORD` 
- - in the Vagrant file `config.registration.username` and `config.registration.password`
- - during the vagrant up in the registration prompt
 
+3. Add the subscription, registry and certificates information :
```
Comment (i386x) [inline README.md:74]: Can Ansible Vault be helpful with handling secrets? https://docs.ansible.com/ansible/latest/vault_guide/index.html

File: roles/tpa_single_node/tasks/podman.yml [hunk 6]
```diff
@@ -3,7 +3,7 @@
   when:
     - ansible_facts['distribution'] == 'RedHat'
     - ansible_facts['distribution_major_version'] == '9'
-  ansible.builtin.command: podman login registry.redhat.io -u {{ tpa_single_node_registry_username }} --password {{ tpa_single_node_registry_password }}
+  ansible.builtin.command: podman login registry.redhat.io -u {{ tpa_single_node_registry_username }} -p {{ tpa_single_node_registry_password }}
```
Comment (i386x) [inline roles/tpa_single_node/tasks/podman.yml:6]: Is there a danger of leaking password to the terminal/logs?

File: README.md (hunk 3 above)
Comment (gildub) [reply README.md:87]: Yes absolutely.

`kubectl` can easily generate TLS secret manifest files from key and certs.
Meanwhile `kubectl` is not part of vanilla RHEL subscription and we don't want to force end user to hold specific subscriptions in order have that tool available.

The other solution is to install kubernetes repo and pull kubectl from it, but would that be acceptable ?

File: README.md (hunk 5 above)
Comment (gildub) [reply README.md:74]: @i386x, that sounds like a good secure way. 

To be discussed with @desmax74 if we want to use it.
Meanwhile that would be taken care in a follow-up PR.

**Comment by i386x** [comment]
Looks good to me, added few notes.

**Review by i386x — APPROVED** [review]

File: roles/tpa_single_node/tasks/podman.yml (hunk 6 above)
Comment (gildub) [reply roles/tpa_single_node/tasks/podman.yml:6]: I believe so when using Ansible along those type of command.
It's better having secret values in envars than in plain text inside configuration files !
This is why using an auth access token (see description) is always safer.

File: README.md (hunk 4 above)
Comment (gildub) [reply README.md:54]: @i386x, good to know. Not sure if that's going to help much for TPA Single Node deployment though.
I could definitely be handy when doing TPA multiple nodes deployments.

**Pull request merged**